testing/trino-tests/src/test/java/io/trino/tests/TestJoinQueries.java

```diff
@@ -42,4 +45,41 @@ public void verifyDynamicFilteringEnabled()
                 "SHOW SESSION LIKE 'enable_dynamic_filtering'",
                 "VALUES ('enable_dynamic_filtering', 'true', 'true', 'boolean', 'Enable dynamic filtering')");
     }
+
+    /**
+     * This test verifies if a broadcast deadlock is getting properly resolved.
+     * <p>
+     * A deadlock can happen when the build side of a join overflows the total capacity of the broadcast output buffer.
+     * When the broadcast buffer is overflow some data must be discarded. The data from the broadcast output buffer can
+     * only be discarded after it is consumed by all consumers. The scheduler is expected to send the "noMoreOutputBuffers"
+     * signal when the probe side scheduling is done. However if the probe side scheduling is blocked on split placement
+     * the scheduling might never finish. To handle this case a special handling was introduced. When the scheduler detects
+     * that the stage is blocked on the split placement and the output buffers of the source tasks of the stage are full the
+     * scheduler schedules as many tasks as there are nodes in the cluster (without waiting for the split placement to finish)
+     * and sends a signal to the source tasks that no more tasks (thus output buffers) will be created.
+     */
+    @Test(timeOut = 600_000)
```

Comment:
That's a really long time -- how long does the test take to run? Let's pick a timeout that's closer to typical execution times -- maybe a 2x or 3x factor, at most.

Reply:
It must create a new query runner, and the time that it takes to create one might vary. It is pretty much a trade off. A higher likelihood of a false timeout vs more time it takes to discover an issue when there is one. Since normally it shouldn't fail It feels like the lower chance of a false negative is more important. What do you think?

Reply:
If this test takes 10 minutes, we have bigger problems...

Reply:
I aggree. Maybe 10 minutes is an extreme. How about ~2 minutes?

Reply:
Decreased to 2 minutes and added a comment